### PR TITLE
remove convert-lora-to-ggml.py

### PR DESCRIPTION
**Pull request by slaren**
Changes such as permutations to the tensors during model conversion makes converting loras from HF PEFT unreliable, so to avoid confusion I think it is better to remove this entirely until this feature is re-evaluated. It is still possible to use loras created with the finetune example.

### CODE REVIEW

**Review by ggerganov — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -1271,18 +1271,7 @@ set_target_properties(llama PROPERTIES PUBLIC_HEADER ${CMAKE_CURRENT_SOURCE_DIR}
...
-        OWNER_WRITE
-        OWNER_EXECUTE
-        GROUP_READ
-        GROUP_EXECUTE
-        WORLD_READ
-        WORLD_EXECUTE
-    DESTINATION ${CMAKE_INSTALL_BINDIR})
-install(
-    FILES convert-lora-to-ggml.py
+    FILES convert.py # does this need to be changed to convert-hf-to-gguf.py?
```
Comment (slaren) [inline CMakeLists.txt:1274]:
```suggestion
    FILES convert.py
```

**Comment by icfly2** [comment]
If I read the examples/finetune right it states that only llama models are supported. And with the `export-lora` I'm not having much luck either with a Phi3 based model that I could previously convert without issues with the removed `.py` script. 

What is the plan going forward?

**Comment by miaoshouai** [comment]
what is the recommanded ways to merge a finetuned lora into base model now?
most of the online documents are refering to use convert-lora-to-ggml and now it is deprecated.

**Comment by slaren** [comment]
You can merge the model using `merge_and_unload` and then convert it to gguf with `convert-hf-to-gguf.py`.

**Comment by rascasoft** [comment]
> > You can merge the model using `merge_and_unload` and then convert it to gguf with `convert-hf-to-gguf.py`.
> 
> For the ones that we dont know very much about this... do you have any document on how it should be the procedure now the script has been removed?

I feel like this question needs to be answered: what does "using `merge_and_unload`" means for someone who have previously just used `convert-lora-to-ggml.py` and now can do this anymore?

**Comment by shreyasrajesh0308** [comment]
I think it means once you have completed training your model, you merge your base model with your adapter weights using the model.merge_and_unload() command in hf. This [thread](https://discuss.huggingface.co/t/help-with-merging-lora-weights-back-into-base-model/40968) has a discussion on it. Hope this helps!

**Comment by rascasoft** [comment]
Thanks, @shreyasrajesh0308 I saw the thread and what I understood is that these things work at Python level, so "code", instead using the script was more "CI friendly", if I'm explaining myself.
Have you got any suggestions/examples about an entire workflow in python usable inside a pipeline?

**Pull request merged**